plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/stock-indicator/block.tsx

```diff
@@ -64,7 +87,7 @@ export const Block = ( props: Props ): JSX.Element | null => {
 	const { parentClassName } = useInnerBlockLayoutContext();
 	const { product } = useProductDataContext();
 
-	if ( ! product.id ) {
+	if ( ! product.id || ! isStockVisible( product ) ) {
```

Comment:
How could it impact the block in a different context? Do you?
Should we check with other stakeholders, or are we safe?